src/pages/Account/Account.tsx

```diff
@@ -2,21 +2,70 @@ import Layout from "../../components/Layout/Layout";
 import "./Account.scss";
 import Header from "../../components/Header/Header";
 import Button from "../../components/Button/Button";
+import { auth } from "../../firebase";
+import { User, signOut } from "firebase/auth";
+import { useNavigate } from "react-router-dom";
+import { getUser } from "../../utils/firebaseSnapshots";
+import { useEffect, useState } from "react";
 import placeHolderPFP from "/src/assets/images/placeHolderPFP.svg";
-import InputBox from "../../components/InputBox/InputBox";
 
-const Account = () => {
-  const handleSignOut = () => {
-    return;
+type AccountProps = {
+  setUser: (user: User | null) => void;
+  user: User;
+};
+
+const Account = ({ setUser, user }: AccountProps) => {
+  const [firstName, setFirstName] = useState<string>("");
+  const [lastName, setLastName] = useState<string>("");
+  const [email, setEmail] = useState<string>("");
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const fetchDisplayName = async () => {
+      const currentUser = await getUser(user.uid);
+
+      if (currentUser) {
+        const firstName = currentUser.firstName;
+        const lastName = currentUser.lastName;
+        const email = currentUser.email;
+        setFirstName(firstName);
+        setLastName(lastName);
+        setEmail(email);
+      }
+    };
+    fetchDisplayName();
+  }, [user]);
+
+  const handleSignOut = async () => {
+    try {
+      await signOut(auth);
+      setUser(null);
+
+      const navigateToSplash = () => {
```

Comment:
I do not think you need to wrap this in a function you should be able to just call `navigate("/splash");`